src/useQueue/__docs__/story.mdx

```diff
@@ -0,0 +1,41 @@
+import { Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
+import { Example } from './example.stories';
+import { ImportPath } from '../../__docs__/ImportPath';
+
+<Meta title="State/useQueue" component={Example} />
+
+# useQueue
+
+A react state hook implements a simple first-in first-out (FIFO) queue.
+
+#### Example
+
+<Canvas>
+  <Story story={Example} inline />
+</Canvas>
+
+## Reference
+
+```ts
+export function useQueue<T[]>(initialValue: T[] =[]): {
+  add: (item: T) => void;
+  remove: () => T;
+  first: T;
+  size: number;
+}
+```
+
+#### Importing
+
+<ImportPath />
+
+#### Arguments
+
+- **initialValue** _`T[]`_ - The initial value of the queue state. Defaults to an empty array.
+
+#### Return
+
+- **first** _`T`_ - The value of the first item in the queue.
+- **last** _`T`_ - The value of the last item in the queue.
+- **add** _`(item: T) => void`_ - Adds a value to the end of the queue.
+- **remove** _`() => T`_ - Removed and returns the value from the end of the queue
```

Comment:
This description seems to be wrong: the value is removed from the beginning, not the end, so this should be: "Removes and returns the head of the queue".

Reply:
Oops 😅